bindings/go/mio/mio.go

```diff
@@ -441,89 +455,121 @@ func getBW(n int, d time.Duration) (int, string) {
     return bw, "Bytes/sec"
 }
 
-func (mio *Mio) Write(p []byte) (n int, err error) {
+func (mio *Mio) write(p []byte, off *int64) (n int, err error) {
     if mio.obj == nil {
         return 0, errors.New("object is not opened")
     }
-    left, off := len(p), 0
+
+    var v iov
+    if err = v.alloc(); err != nil {
+        return 0, err
+    }
+    defer v.free()
+
+    left := len(p)
     bs, gs := mio.getOptimalBlockSz(left)
-    start, offSaved, bsSaved := time.Now(), mio.off, bs
+    start, offSaved, bsSaved := time.Now(), *off, bs
     for ; left > 0; left -= bs {
         if left < bs {
             bs = left
         }
-        slot := <-mio.ch // get next available from the pool
+        slot := <-v.ch // get next available from the pool
         if slot.err != nil {
             break
         }
-        err = mio.prepareBuf(p, slot.idx, bs, gs, off, mio.off)
+        err = v.prepareBuf(p[n:], slot.idx, bs, gs, *off)
         if err != nil {
-            return off, err
+            return n, err
         }
-        if mio.minBuf != nil {
-            copy(mio.minBuf, p[off:])
+        if v.minBuf != nil { // last block, not aligned
+            copy(v.minBuf, p[n:])
         }
-        mio.wg.Add(1)
-        go mio.doIO(slot.idx, C.M0_OC_WRITE)
-        off += bs
-        mio.off += uint64(bs)
+        v.wg.Add(1)
+        go v.doIO(mio.obj, slot.idx, C.M0_OC_WRITE)
+        n += bs
```

Comment:
The return value `n` is automatically initialized to "zero" by the go language?

Reply:
Yes. Refer to https://tour.golang.org/basics/12.